### PR TITLE
Update versions of OS, JDK, BuildTools, etc

### PR DESCRIPTION
**Pull request by radetsky**
I successfully built an android-themis on the image.

### CODE REVIEW

**Review by vixentael — APPROVED** [review]
are you sure we should merge to master? maybe to 0.15.0 release branch as well?

**Comment by radetsky** [comment]
Sure. It is not Themis repository. There is no sense in making a dedicated branch for each release for cossacklabs/dockerfiles.

**Comment by vixentael** [comment]
ahhh sorry, you are right!

File: android-build/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
-FROM ubuntu:22.04
+FROM ubuntu:23.04
```
Comment (Lagovas) [inline android-build/Dockerfile]: can you explain why we need not LTS version of ubuntu? is there any reason why 22.04 is not enough?

Comment (radetsky) [reply android-build/Dockerfile]: Mea culpa. Fixed. Tested.

File: android-build/Dockerfile [hunk 2]
```diff
@@ -37,8 +37,8 @@ RUN wget -O ~/cmdline-tools-linux.zip \
     ~/android-sdk/cmdline-tools/latest/bin/sdkmanager \
         'build-tools;34.0.0' \
         'cmake;3.22.1' \
-        'platforms;android-31' \
-        'ndk;26.0.10792818'
+        'platforms;android-33' \
+        'ndk;25.0.8775105'
```
Comment (vixentael) [inline android-build/Dockerfile:41]: we are decreasing ndk version, but increasing android version?  okoko

**Review by vixentael — APPROVED** [review]

**Pull request merged**